gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -171,10 +236,84 @@ components:
+          type: string
+          nullable: false
+        allowOverride:
+          description: >
+            By default, the task is reassigned if it was already assigned. If this option is set
+            to false then an error will be returned in such cases. The task must then first be
+            unassigned in order to be assigned again.
+          type: boolean
+          nullable: true
+        action:
+          description: >
+            A custom action value that will be accessible from user task events resulting
+            from this endpoint invocation. If not provided, it will default to "assign".
+          type: string
+          nullable: true
+    UserTaskUpdateRequest:
+      type: object
+      properties:
+        changeset:
+          description: |
+            JSON object with changed task attribute values.
+
+            The following attributes can be adjusted with this endpoint, additional attributes
+            will be ignored:
+
+            * `candidateGroups` - reset by providing an empty list
+            * `candidateUsers` - reset by providing an empty list
+            * `dueDate` - reset by providing an empty String
+            * `followUpDate` - reset by providing an empty String
+
+            Providing any of those attributes with a `null` value or omitting it preserves
+            the persisted attribute's value.
+
+            The assignee cannot be adjusted with this endpoint, use the Assign task endpoint.
+            This ensures correct event emission for assignee changes.
+          type: object
+          nullable: true
+          $ref: "#/components/schemas/Changeset"
+        action:
+          description: >
+            A custom action value that will be accessible from user task events resulting
+            from this endpoint invocation. If not provided, it will default to "update".
+          type: string
+          nullable: true
     Variables:
       description: A map of variables.
       type: object
       additionalProperties: true
+    Changeset:
+      description: A map of changes.
+      type: object
+      additionalProperties: true
+      properties:
+        dueDate:
+          type: string
+          description: The due date of the task. Reset by providing an empty String.
+          nullable: true
+        followUpDate:
+          type: string
+          description: The follow-up date of the task. Reset by providing an empty String.
+          nullable: true
+        candidateUsers:
+          type: array
+          description: The list of candidate users of the task. Reset by providing an empty list.
+          items:
+            type: string
+          nullable: true
+        candidateGroups:
+          type: array
+          description: The list of candidate groups of the task. Reset by providing an empty list.
+          items:
+            type: string
+          nullable: true
```

Comment:
🕵️ Yes, that's correct. I am also still wondering if that makes sense or not 😄 Currently, it even creates a command for the broker. Maybe we can prevent this to avoid unnecessary processing.

Reply:
I adjusted the initial description to include a 400 when neither of them was provided.

Reply:
> 🕵️ Yes, that's correct. I am also still wondering if that makes sense or not 😄 Currently, it even creates a command for the broker. Maybe we can prevent this to avoid unnecessary processing.

This is what I was thinking too. I wonder if it would be a good idea to return a response from controller if the request body is empty?

Reply:
Yep, tackled this with https://github.com/camunda/zeebe/pull/16537/commits/71a08472e67c302407370b3ff201180c4025371d.